### ISSUE TITLE
_tkmktap and _websetroot are empty.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3074]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#3074|
|Type|enhancement|
|Created|2008-03-12 17:56:21Z|
|Branch|https://github.com/twisted/twisted/tree/zsh-stubs-3074|

A couple of files in twisted/python/zsh are empty.

I hear this stuff is rather poorly maintained and that websetroot doesn't even exist any more.

Someone should clean this up, or remove the zsh noise if it isn't maintained.

<details><summary>Searchable metadata</summary>

```
trac-id__3074 3074
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__low low
milestone__ 
branch__branches_zsh_stubs_3074 branches/zsh-stubs-3074
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1205344581000000 1205344581000000
changetime__1307069096000000 1307069096000000
version__None None
owner__exarkun exarkun
cc__jesstess cc__khorn cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @teratorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_2"></a>|@jesstess commented|
|-|-|

This is a note in twisted/python/zshcomp.py

# NOTE: the commands using None above are no longer included in Twisted. However due to limitations in zsh's completion system the version of _twisted_zsh_stub shipped with zsh contains a static list of Twisted's commands. It will display errors if completion functions for these missing commands are not found :( So we just include dummy (empty) completion function files.

So it looks like these files are supposed to exist and be empty, unless that note is out of date.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_3"></a>|@khorn commented|
|-|-|

Would it be possible to have the files include some kind of comment to indicate that their existence (and emptiness) is required?

I don't know enough about this topic to know whether there's any way of doing this without screwing up the intended effect (command completion).

If there's no way to put anything in the files themselves, perhaps an explanation could be added to `twisted\python\zsh\README`?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

> the version of _twisted_zsh_stub shipped with zsh contains a static list of Twisted's commands

First, someone should contact the zsh people and tell them that the command list has changed.  They need to update the file they're distributed, or perhaps stop distributing it altogether.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

I reported this to zsh, <https://sourceforge.net/tracker/?func=detail&aid=3305925&group_id=4068&atid=104068>.  I think we should just delete these files (and perhaps all the rest of our zsh completion related stuff - but that's another ticket).  If zsh doesn't care enough to fix the issue in a timely manner, then I don't know why we should (because, hey, I don't use zsh, and apparently teratorn was the only Twisted user who cared enough to work on this thing).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [31927]) Branching to 'zsh-stubs-3074'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun removed owner|
|-|-|

Removed them

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [32019]) Merge zsh-stubs-3074

Author: exarkun
Reviewer: thijs
Fixes: [#3074](https://github.com/twisted/twisted/issues/3074)

Delete the empty `_tkmktap` and `_websetroot` zsh completion stubs.  The
corresponding commands no longer exist.
